test/test_convergence.jl

```diff
@@ -0,0 +1,213 @@
+thermo_params = SFP.thermodynamics_params(param_set)
+uft = SFP.universal_func_type(param_set)
+
+"""
+    PhaseEquilProfiles(param_set, ::Type{ArrayType})
+
+Returns a `ProfileSet` used to test moist states in thermodynamic equilibrium.
+"""
+function PhaseEquilProfiles(param_set::APS, ::Type{ArrayType}) where {ArrayType}
+    phase_type = TD.PhaseEquil
+
+    # Prescribe z_range, relative_sat, T_surface, T_min
+    z_range, relative_sat, T_surface, T_min = input_config(ArrayType)
+
+    # Compute T, p, from DecayingTemperatureProfile, (reshape RS)
+    z, T_virt, p, RS = shared_profiles(param_set, z_range, relative_sat, T_surface, T_min)
+    T = T_virt
+
+    FT = eltype(T)
+    R_d::FT = CPP.R_d(param_set)
+    grav::FT = CPP.grav(param_set)
+    # Compute total specific humidity from temperature, pressure
+    # and relative saturation, and partition the saturation excess
+    # according to temperature.
+    ρ = p ./ (R_d .* T)
+    q_tot = RS .* TD.q_vap_saturation.(Ref(param_set), T, ρ, Ref(phase_type))
+    q_pt = TD.PhasePartition_equil.(Ref(param_set), T, ρ, q_tot, Ref(phase_type))
+
+    # Extract phase partitioning and update pressure
+    # to be thermodynamically consistent with T, ρ, q_pt
+    q_liq = getproperty.(q_pt, :liq)
+    q_ice = getproperty.(q_pt, :ice)
+    p = TD.air_pressure.(Ref(param_set), T, ρ, q_pt)
+
+    e_int = TD.internal_energy.(Ref(param_set), T, q_pt)
+    θ_liq_ice = TD.liquid_ice_pottemp.(Ref(param_set), T, ρ, q_pt)
+    RH = TD.relative_humidity.(Ref(param_set), T, p, Ref(phase_type), q_pt)
+    e_pot = grav * z
+    Random.seed!(15)
+    u = rand(FT, size(T)) * 5
+    v = rand(FT, size(T)) * 5
+    w = rand(FT, size(T)) * 5
+    e_kin = (u .^ 2 .+ v .^ 2 .+ w .^ 2) / 2
+
+    return ProfileSet{typeof(T), typeof(q_pt), typeof(phase_type)}(
+        z,
+        T,
+        p,
+        RS,
+        e_int,
+        ρ,
+        θ_liq_ice,
+        q_tot,
+        q_liq,
+        q_ice,
+        q_pt,
+        RH,
+        e_pot,
+        u,
+        v,
+        w,
+        e_kin,
+        phase_type,
+    )
+end
+function input_config(ArrayType; n = 10, n_RS1 = 20, n_RS2 = 20, T_surface = 290, T_min = 150)
+    n_RS = n_RS1 + n_RS2
+    z_range = ArrayType(range(0, stop = 5e2, length = n))
+    relative_sat1 = ArrayType(range(0, stop = 1, length = n_RS1))
+    relative_sat2 = ArrayType(range(1, stop = 1.02, length = n_RS2))
+    relative_sat = vcat(relative_sat1, relative_sat2)
+    return z_range, relative_sat, T_surface, T_min
+end
+
+"""
+    shared_profiles(
+        param_set::APS,
+        z_range::AbstractArray,
+        relative_sat::AbstractArray,
+        T_surface,
+        T_min,
+    )
+
+Compute profiles shared across `PhaseDry`,
+`PhaseEquil` and `PhaseNonEquil` thermodynamic
+states, including:
+ - `z` altitude
+ - `T` temperature
+ - `p` pressure
+ - `RS` relative saturation
+"""
+function shared_profiles(param_set::APS, z_range::AbstractArray, relative_sat::AbstractArray, T_surface, T_min)
```

Comment:
Why was this defined here? Couldn't we just instead call the function in Thermodynamics rather than duplicating this function? cc @szy21, @akshaysridhar

Reply:
We actually wanted to change the thermodynamic states generated but didn't find a way to do so. We plan to change this shortly, in a separate PR, and @akshaysridhar knows more. Could we chat about this if you have some time?

Reply:
But, this code is exactly the same as what's in Thermodynamics.jl? Sure, we should chat

Reply:
Fixing in #57